esp-hal-common/Cargo.toml

```diff
@@ -19,8 +19,9 @@ critical-section     = "1.1.2"
 embedded-can         = { version = "0.4.1", optional = true }
 embedded-dma         = "0.2.0"
 embedded-hal         = { version = "0.2.7", features = ["unproven"] }
-embedded-hal-1       = { version = "=1.0.0-alpha.11", optional = true, package = "embedded-hal" }
-embedded-hal-nb      = { version = "=1.0.0-alpha.3", optional = true }
+embedded-hal-1       = { version = "=1.0.0-rc.1", optional = true, package = "embedded-hal" }
+embedded-hal-nb      = { version = "=1.0.0-rc.1", optional = true }
+embedded-io          = "0.5.0"
```

Comment:
shouldn't this be optional and only get included if we target eh1 / async?

Reply:
The `embedded_io` trait implementations are not feature gated currently. Should they be? I didn't see any reason to, they're not really following the `1.0.0-xxx` release cycle of the other packages. But I can change this if there's a reason to.

Reply:
well we pull an unused dependency this way 🤷‍♂️

Reply:
`embedded-hal` and `embedded-dma` should both be optional too then, right? I'm not really sure where we're drawing the line here.

Reply:
I think code implementing embedded-hal and embedded-dma is not feature gated.
But embedded-hal-1 is and it's optional because of that.